internal/monitors/disk/disk.go

```diff
@@ -0,0 +1,191 @@
+	m.Output.SendDatapoint(datapoint.New("disk_octets.avg_write", dimensions, datapoint.NewIntValue(int64(disk.WriteBytes)), datapoint.Gauge, time.Time{}))
+
+	// disk_merged.read - N/A
+	// disk_merged.write - N/A
+
+	// disk_time.read
+	m.Output.SendDatapoint(datapoint.New("disk_time.avg_read", dimensions, datapoint.NewIntValue(int64(disk.ReadTime)), datapoint.Gauge, time.Time{}))
+
+	// disk_time.write
+	m.Output.SendDatapoint(datapoint.New("disk_time.avg_write", dimensions, datapoint.NewIntValue(int64(disk.WriteTime)), datapoint.Gauge, time.Time{}))
+}
+
+func (m *Monitor) processLinuxDatapoints(disk *gopsutil.IOCountersStat, dimensions map[string]string) {
+	// disk_ops.read
+	m.Output.SendDatapoint(datapoint.New("disk_ops.read", dimensions, datapoint.NewIntValue(int64(disk.ReadCount)), datapoint.Counter, time.Time{}))
+
+	// disk_ops.write
+	m.Output.SendDatapoint(datapoint.New("disk_ops.write", dimensions, datapoint.NewIntValue(int64(disk.WriteCount)), datapoint.Counter, time.Time{}))
+
+	// disk_octets.read
+	m.Output.SendDatapoint(datapoint.New("disk_octets.read", dimensions, datapoint.NewIntValue(int64(disk.ReadBytes)), datapoint.Counter, time.Time{}))
+
+	// disk_octets.write
+	m.Output.SendDatapoint(datapoint.New("disk_octets.write", dimensions, datapoint.NewIntValue(int64(disk.WriteBytes)), datapoint.Counter, time.Time{}))
+
+	// disk_merged.read
+	m.Output.SendDatapoint(datapoint.New("disk_merged.read", dimensions, datapoint.NewIntValue(int64(disk.MergedReadCount)), datapoint.Counter, time.Time{}))
+
+	// disk_merged.write
+	m.Output.SendDatapoint(datapoint.New("disk_merged.write", dimensions, datapoint.NewIntValue(int64(disk.MergedWriteCount)), datapoint.Counter, time.Time{}))
+
+	// disk_time.read
+	m.Output.SendDatapoint(datapoint.New("disk_time.read", dimensions, datapoint.NewIntValue(int64(disk.ReadTime)), datapoint.Counter, time.Time{}))
+
+	// disk_time.write
+	m.Output.SendDatapoint(datapoint.New("disk_time.write", dimensions, datapoint.NewIntValue(int64(disk.WriteTime)), datapoint.Counter, time.Time{}))
+}
+
+func shouldSkipDisk(conf *Config, disk *gopsutil.IOCountersStat) (shouldSkip bool) {
+	_, match := conf.ignoreString[disk.Name]
+	match = match || utils.FindMatchString(disk.Name, conf.ignoreRegex)
+	if (match && *conf.IgnoreSelected) || (!match && !*conf.IgnoreSelected) {
+		shouldSkip = true
+	}
+	return
+}
+
+// EmitDatapoints emits a set of memory datapoints
+func (m *Monitor) emitDatapoints(conf *Config) {
+	iocounts, err := iOCounters()
+	if err != nil {
+		logger.WithError(err).Warningf("failed to load io counters. if this message repeates frequently there may be a problem")
+	}
+	// var total uint64
+	for key, disk := range iocounts {
+		// handle selecting disk
+		if shouldSkipDisk(conf, &disk) {
+			logger.Debugf("skipping disk '%s'", disk.Name)
+			continue
+		}
+
+		pluginInstance := strings.Replace(key, " ", "_", -1)
+		dimensions := map[string]string{"plugin": monitorType, "plugin_instance": pluginInstance}
```

Comment:
This is fine if we keep `plugin_instance` for content purposes, but let us at least duplicate the dimension to something that actually tells us what the vaule is supposed to be, like `disk` or something.

Reply:
ack